server/service/integration_core_test.go

```diff
@@ -314,6 +314,27 @@ func (s *integrationTestSuite) TestAppConfigAdditionalQueriesCanBeRemoved() {
 	assert.True(t, config.HostExpirySettings.HostExpiryEnabled)
 }
 
+func (s *integrationTestSuite) TestAppConfigDetailQueriesOverrides() {
+	t := s.T()
+
+	spec := []byte(`
+  features:
+    additional_queries:
+      time: SELECT * FROM time
+    enable_host_users: true
+    detail_query_overrides:
+      users: null
+      software_linux: "select * from blah;"
+`)
+	s.applyConfig(spec)
+
+	config := s.getConfig()
+	require.NotNil(t, config.Features.DetailQueryOverrides)
+	require.Nil(t, config.Features.DetailQueryOverrides["users"])
+	require.NotNil(t, config.Features.DetailQueryOverrides["software_linux"])
+	require.Equal(t, "select * from blah;", *config.Features.DetailQueryOverrides["software_linux"])
+}
+
```

Comment:
feels like there's a missing test here (or in `integration_live_queries_test.go` which despite the name tests queries in general) where we change the config and `/api/osquery/distributed/read` returns the right detail queries.

Reply:
Good point. Added a test. [60d3e72](https://github.com/fleetdm/fleet/pull/9117/commits/60d3e72d9c4ee0c4d7ed4a58f713bba420067ac4).